contrib/zbm-sign.pl

```diff
@@ -72,17 +76,14 @@
   if ( $SignMethod eq "sbctl" ) {
     system "sbctl sign $ZBM/$_";
   } elsif ( $SignMethod eq "sbsign" ) {
-    $Unsigned = substr( $_, 0, -4 );
-    system "sbsign --key $KeyDir/DB.key --cert $KeyDir/DB.crt $ZBM/$_ --output $ZBM/$Unsigned-signed.efi";
+    my $verify_output = "sbverify --cert $KeyDir/$CrtFileName $ZBM/$_ 2>&1";
+    if ( $verify_output =~ /Signature verification OK/ ) {
+      say "File $_ is already signed.";
+      next;
+    }
+    system "sbsign --key $KeyDir/$KeyFileName --cert $KeyDir/$CrtFileName $ZBM/$_ --output $ZBM/$_";
   } else {
     die "Sign method $SignMethod not valid.";
   }
-
-  if ( $DeleteUnsigned && $SignMethod eq "sbctl" ) {
```

Comment:
Why was this whole block removed? It seems like it will fundamentally change the behavior of this script.

Reply:
Really? It does only 1 of the 2 things:

1. Print a message if `sbctl` is used
2. Clean up after the inability of `sbsign` to sign in place... with the new code, that is no longer the case

Maybe I should remove the option `DeleteUnsigned` since it is no longer needed.

What am I missing? I see this block of code as no longer needed.

Reply:
Are you sure that sbsign won't corrupt its input if you try to overwrite it with output? Is that the case for several prior versions of sbsign?

If it's safe, the better approach would be to replace `DeleteUnsigned` with something a bit more meaningful like `Overwrite` and handle that above by just changing what you're passing as the output name to sbsign.

Reply:
Then I propose `UseOriginalFileName`:
- true -> *.efi = stays the same.
- false (default) -> *.signed.efi

Or you can think of a better boolean var name?

Reply:
I proposed, and still prefer, `Overwrite`, which is both shorter and more descriptive.